src/components/ToggleSwitch.js

```diff
@@ -49,125 +49,168 @@ export default class ToggleSwitch extends React.Component<ToggleSwitchDefaultPro
+        }
+      }
+      buttonStyle = {
+        boxShadow: '0px 2px 2px rgba(0, 0, 0, 0.24), 0px 0px 2px rgba(0, 0, 0, 0.12)',
+        height: '17px',
+        width: '17px',
+        borderRadius: '100%',
+        position: 'relative',
+        transitionDuration: '100ms',
+        zIndex: '1',
+        cursor: disabled ? 'default' : 'pointer',
+        outline: 'none',
+        border: 'none',
+        transition: 'right 300ms cubic-bezier(0.26, 0.86, 0.44, 0.98)',
+        ...buttonStyle,
+      };
+    }
+
+    return {
+      button: {
+        className: buttonClassName,
+        style: buttonStyle,
+      },
+      slider: {
+        className: sliderClassName,
+        style: sliderStyle,
+      },
+      container: {
+        className: containerClassName,
+        style: containerStyle,
+      },
+      onClassName,
+      offClassName,
+    };
+  }
 
-      const sliderClassName = on && !disabled
-        ? 'attivio-primary-background'
-        : 'attivio-gray-light-background';
+  labeledOnButton: ?HTMLDivElement;
+  labeledOffButton: ?HTMLDivElement;
 
-      return (
+  renderNoLabel() {
+    const { container, slider, button } = this.getStyles();
+
+    return (
+      <div
+        className={container.className}
+        style={container.style}
+      >
         <div
-          className={containerClass}
-          style={{
-            width: '100%',
-            height: '100%',
-            display: 'flex',
-            alignItems: 'center',
-            justifyContent: 'center',
-          }}
-        >
-          <div
-            className={sliderClassName}
-            style={{
-              width: '28px',
-              height: '12px',
-              borderRadius: '10px',
-              opacity: '0.7',
-            }}
-          />
-          <div
-            className={buttonClassName}
-            onClick={() => {
-              if (!on && !disabled) {
-                // Off now, turn it on
-                onChange(true);
-              }
-              if (on && !disabled) {
-                // On now, turn it off
-                onChange(false);
-              }
-            }}
-            role="button"
-            tabIndex={0}
-            style={{
-              boxShadow: '0px 2px 2px rgba(0, 0, 0, 0.24), 0px 0px 2px rgba(0, 0, 0, 0.12)',
-              height: '17px',
-              width: '17px',
-              borderRadius: '100%',
-              position: 'relative',
-              transitionDuration: '100ms',
-              zIndex: '1',
-              cursor: disabled ? 'default' : 'pointer',
-              outline: 'none',
-              border: 'none',
-              transition: 'right 300ms cubic-bezier(0.26, 0.86, 0.44, 0.98)',
-              ...buttonStyle,
-            }}
-          />
-        </div>
-      );
-    }
+          className={slider.className}
+          style={slider.style}
+        />
+        <div
+          className={button.className}
+          onClick={this.onClick}
```

Comment:
One thing I noticed when poking around in the UI was that for the no-label variation, you have to click the circle to toggle the switch. I think it would be useful to be able to click anywhere on the component to do this... which could be done by just moving the onClick to the outer `<div>`, I think. Thoughts?

Reply:
makes sense, I'll change it